src/Sylius/Bundle/ApiBundle/EventSubscriber/CatalogPromotionEventSubscriber.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\EventSubscriber;
+
+use ApiPlatform\Core\EventListener\EventPriorities;
+use Sylius\Component\Core\Model\CatalogPromotionInterface;
+use Sylius\Component\Promotion\Event\CatalogPromotionUpdated;
+use Sylius\Component\Promotion\Model\CatalogPromotionActionInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\ViewEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\Messenger\MessageBusInterface;
+
+final class CatalogPromotionEventSubscriber implements EventSubscriberInterface
+{
+    private MessageBusInterface $eventBus;
+
+    public function __construct(MessageBusInterface $eventBus)
+    {
+        $this->eventBus = $eventBus;
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            KernelEvents::VIEW => ['postWrite', EventPriorities::POST_WRITE],
+        ];
+    }
+
+    public function postWrite(ViewEvent $event): void
+    {
+        $entity = $event->getControllerResult();
+        $method = $event->getRequest()->getMethod();
+
+        if ($method === Request::METHOD_PUT || $method === Request::METHOD_PATCH) {
+            if ($entity instanceof CatalogPromotionInterface) {
+                $this->eventBus->dispatch(new CatalogPromotionUpdated($entity->getCode()));
+            }
+
+            if ($entity instanceof CatalogPromotionActionInterface) {
```

Comment:
Is this if needed? I guess that after updating a catalog promotion action on catalog promotion endpoint, we need to subscribe only on changes on `CatalogPromotionInterface`